core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -289,7 +289,7 @@ private[log] object LogCleanerManager extends Logging {
     }
 
     // dirty log segments
-    val dirtyNonActiveSegments = log.logSegments(firstDirtyOffset, log.activeSegment.baseOffset).toArray
+    val dirtyNonActiveSegments = log.logSegments(firstDirtyOffset, log.activeSegment.baseOffset)
```

Comment:
Should we slap a `toIndexSeq` here as well, just to avoid further attempts to use `toArray` or is it OK to leave it as is, considering it's only being used to `find` a segment right now which doesn't depend on constant time access?